app/controllers/RestaurantController.java

```diff
@@ -75,11 +81,26 @@ public Result deleteRestaurant(String id) {
 	@Transactional(readOnly = true)
 	public Result getAllRestaurants() {
 		String cityFilter = request().getQueryString(CITY_FILTER);
+		String queryString = request().uri();			
+		String cuisineFilter;
+ 		Pattern pattern = Pattern.compile("cuisineFilter=([\\d\\D\\,]*)&");
+    	Matcher matcher = pattern.matcher(queryString);
```

Comment:
Simply `String cuisineFilter = request().getQueryString("cuisineFilter")` would do the same.

Reply:
Correction: `String cuisineFilter = request().queryString.get("cuisineFilter")`